core/ignore.js

```diff
@@ -139,16 +140,28 @@ class Ignore {
   }
 
   // Return true if the given file/folder path should be ignored
-  isIgnored (doc /*: Metadata */) {
+  isIgnored (doc /*: Metadata | AtomWatcherEvent */) {
```

Comment:
`isIgnored` returns `true` if the given file/folder path should be ignored.
The implementation is changed to accept an event that must contain properties `kind`, and `_id` or `path`.
Then the function gains more responsibilities:
- extract the path from the object parameter
- check registered patterns to ignore or not the path

`isIgnored` must accept a path and an array of patterns and returns `true` or `false`.
The `Ignore' class is instantiated when we want to ignore a path based on a list of patterns, whether that path comes from a `Metadata` object or an `AtomWatcherEvent` that both contain a path property.

This class and its responsabilities are already tested in https://github.com/cozy-labs/cozy-desktop/blob/master/test/unit/ignore.js 

This PR aims to add a test for these lines:
```
module.exports = function (buffer /*: Buffer */, opts /*: { ignore: Ignore } */) /*: Buffer */ {
  return buffer.map((batch) => {
    return batch.filter(event => !opts.ignore.isIgnored(event))
  })
}
```
that takes a buffer and returns a new buffer without ignored events.

This seems a lot of code to do that.

Reply:
I see what you mean. I'll change the tests to use `sinon` stubs.

I'll make another PR to change the implementation of the `isIgnored` method so it expects a `path` and a `kind` (since we'll move to the new watcher, I think it's better to use the new `Kind` type rather than `DocType`) and change the step accordingly.

Reply:
I would have done it in the same PR since this PR needs it and it has not been done in the PR that introduces this need in `core/local/steps'.

For tracking: #1383